types/decimal.go

```diff
@@ -446,20 +450,69 @@ func (d Dec) ApproxRoot(root uint64) (guess Dec, err error) {
 		return OneDec(), nil
 	}
 
-	guess, delta := OneDec(), OneDec()
+	delta := OneDec()
 
-	for iter := 0; delta.Abs().GT(SmallestDec()) && iter < maxApproxRootIterations; iter++ {
+	// here we do a rough estimation of bit lenght the root value should have
+	rootF := float64(root)
+	dBitLen := float64(d.i.BitLen())
+	guessBitLen := int(dBitLen/rootF + 18*((rootF-1)/rootF)*log2of10)
```

Comment:
I'm a bit worried about using direct floats here. @ValarDragon do you think this is safe and deterministic across architectures?

Reply:
there's a ceiling at the end so I guess It should be fine

Reply:
No, lets please not use floats here

Reply:
I'll change this to sdk.dec